src/AppInitializer.tsx

```diff
@@ -247,4 +247,4 @@ class AppInitializer extends React.Component<Props, AppInitializerState> {
   }
 }
 
-export default withRouter(withCookies(AppInitializer))
+export default withLocation(withCookies(AppInitializer))
```

Comment:
withRouter is removed in react-router v6 and AppInitializer is a class component, so it can't use `useLocation`. Created a small wrapper `withLocation` that passes the location as a prop using the `useLocation` hook, which unblocks upgrading to v6.